internal/cli/service_test.go

```diff
@@ -129,95 +304,417 @@ var _ = Describe("CLI Service", func() {
 					}
 				})
 
-				It("instructs the API client to target the specified task key", func() {})
+				JustBeforeEach(func() {
+					_, err := service.InitiateRun(runConfig)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				It("sends the file contents to cloud", func() {
+					Expect(receivedSpecifiedFileContent).To(Equal(originalSpecifiedFileContent))
+				})
 			})
 		})
 
-		Context("with a specific mint directory", func() {
-			BeforeEach(func() {
-				runConfig.MintDirectory = "test"
+		Context("with no specific mint file and no specific directory", func() {
+			Context("when a directory with files is found", func() {
+				var originalMintDirFileContent string
+				var receivedMintDirFileContent string
+
+				BeforeEach(func() {
+					originalMintDirFileContent = "tasks:\n  - key: mintdir\n    run: echo 'mintdir'\n"
+					receivedMintDirFileContent = ""
+					runConfig.MintFilePath = ""
+					runConfig.MintDirectory = ""
+
+					mockFS.MockGetwd = func() (string, error) {
+						return "/some/path/to/working/directory", nil
+					}
+					mockFS.MockExists = func(name string) (bool, error) {
+						return name == "/some/path/to/.mint", nil
+					}
+					mockFS.MockReadDir = func(name string) ([]fs.DirEntry, error) {
+						Expect(name).To(Equal("/some/path/to/.mint"))
+						return []fs.DirEntry{
+							mocks.DirEntry{FileName: "mintdir-tasks.yml"},
+							mocks.DirEntry{FileName: "mintdir-tasks.json"},
+						}, nil
+					}
+					mockFS.MockOpen = func(name string) (fs.File, error) {
+						if name == "/some/path/to/.mint/mintdir-tasks.yml" {
+							file := mocks.NewFile(originalMintDirFileContent)
+							return file, nil
+						} else {
+							Expect(name).To(BeNil())
+							return nil, errors.New("file does not exist")
+						}
+					}
+					mockAPI.MockInitiateRun = func(cfg api.InitiateRunConfig) (*api.InitiateRunResult, error) {
+						Expect(cfg.TaskDefinitions).To(HaveLen(1))
+						// note that this is relative to cwd and that the mint dir files are in task definitions instead of mint directory
+						Expect(cfg.TaskDefinitions[0].Path).To(Equal("../../.mint/mintdir-tasks.yml"))
+						Expect(cfg.MintDirectory).To(BeNil())
```

Comment:
Why is this nil? Shouldn't it have the yaml file in `/some/path/to/.mint`? Maybe something to do with the test setup, but the MockOpen looks right to me..?

Reply:
In the case where no file was specified, we set `TaskDefinitions` to whatever is in the mint directory we find instead of `MintDirectory`.

If Mint gets no mint directory, it assumes whatever was in `TaskDefinitions` is also the mint directory.

I can populate both, but this felt preferable because populating both means we'd have to parse the same thing twice

Reply:
So you'll see this consistently in the contexts which are "no file specified"

Reply:
Hm- actually now that I'm thinking about it more. I might just need to always send it anyway because of the path shenanigans we're doing. The mint directory contents need to be `.mint/`

Reply:
I'll push a commit

Reply:
Got it. I figured that was the behavior, but it seemed odd to me given the last bullet point in the PR description. +1 to sending the mint directory always, even if it's redundant.